### ISSUE TITLE
Best way to merge only last 30 seconds of event?

### ISSUE THREAD

**Issue by mluggy** [ehendrix23/tesla_dashcam#178]
Other than loading the event.timestamp, deducting 30 seconds and passing --start_timestamp

**Comment by GoodyGizmos** [comment]
@mluggy Does this get you close since you would know the typical length of the clips, so the offset in your script could be fairly consistent?

https://github.com/ehendrix23/tesla_dashcam#event-offsets

**Comment by mluggy** [comment]
some clips are 5-10 minutes, some are 1 minute (i.e if you just started driving).

**Comment by mluggy** [comment]
Thanks. Sentry might need to be longer, so 60 seconds from event.

**Comment by mluggy** [comment]
so, any alternative? i think start-timestamp is either not working correctly or conflicts with tesla's/rpi timezone.

**Comment by ehendrix23** [comment]
start-timestamp really is to just set a start timestamp from which to create the videos. For example, if you have events from September 1 till September 22nd and you only want to create starting from September 15th then you provide the start-timestamp for September 15th and it will skip everything before.
Similar with end-timestamp.

For sentry events we calculate the timestamps from the offsets as follow:
```
start timestamp = event start timestamp + start_offset
end timestamp = event start timestamp + end_offset
```
```   --start_offset=0 --end_offset=60 ```
That will result in the starting timestamp for the clip being set to the event timestamp (event timestamp + start_offset) and the ending timestamp for the clip to the event time timestamp + 60 seconds (event timestamp + end_offset).

Now, if you for example want 30 seconds before the event starting timestamp and 60 seconds after then use:
```   --start_offset=-30 --end_offset=60 ```

For non-sentry events however right now it is done a bit different as I take the absolute value:
```
start timestamp = clip start timestamp + absolute value of start_offset
end timestamp = clip start timestamp - absolute value of end_offset
```

So your request for non-Sentry will not work as it is right now.

I think that the logic for sentry events is good, but maybe the logic for non-sentry events should be:
```
start_offset > 0:
      start timestamp = clip start timestamp + start_offset
start_offset < 0:
    start timestamp = clip end timestamp - start_offset

end_offset > 0:
      end timestamp = clip start timestamp + end_offset
end_offset < 0:
    end timestamp = clip end timestamp - end_offset
```
Then you could use start_offset=-30 to get the last 30 seconds. If by giving both start_offset and end_offset it results in end_timestamp being before start_timestamp then we swap both.

How does that sound?

**Comment by mluggy** [comment]
Sounds great!

**Comment by ehendrix23** [comment]
OK, changed it a bit more by including --sentry_start_offset and --sentry_end_offset allowing one to set the offsets separately between sentry events with a event timestamp and others.

To only get the last 30 seconds for non-sentry, and 60 seconds from the start of the sentry event you would use:
```--start_offset=-30 --sentry_start_offset=0 --sentry_end_offset=60```
Reason for ```--sentry_start_offset=0``` is because if it is not provided then it will use ```--start_offset=-30``` for the start of the event based on end timestamp.
See the "Event offsets" section in the readme for more explanation: https://github.com/ehendrix23/tesla_dashcam/blob/dev/README.rst

Note, this is currently not yet released and only in the dev branch. Not sure how you run it (executable, Python) but if Python then you can just download tesla_dashcam.py from the dev branch.
If executable and you want to try it then make sure you have Python installed and then you can download tesla_dashcam.py from the dev branch.

There is another change I want to get in (open PR) and hopefully also a fix for something (open issue) before releasing v0.1.20.

**Comment by ehendrix23** [comment]
My apologies it ended up taking this long, other priorities took precedent. Just released v0.1.20 which includes this requested option.

**Comment by mluggy** [comment]
cool. thanks. project is embedded in the beta version of my product: teslarpi.com
